btk/measure.py

```diff
@@ -135,18 +137,107 @@ def basic_measure(
     **kwargs,
 ):
-    """Return detection, segmentation and deblending information with SEP.
+    """Returns centers detected with source extractor by combining predictions in different bands.
+
+    NOTE: If this function is used with the multiresolution feature,
+    measurements will be carried on the first survey.
+
+    Args:
+        batch (dict): Output of DrawBlendsGenerator object's `__next__` method.
+        idx (int): Index number of blend scene in the batch to preform
+            measurement on.
+        sigma_noise (float): Sigma threshold for detection against noise.
+        matching_threshold (float): Match centers of objects that are closer than
+            this threshold to a single prediction.
+
+    Returns:
+            dict containing catalog with entries corresponding to measured peaks.
+    """
+    channel_indx = 0 if not channels_last else -1
+    # multiresolution
+    if is_multiresolution:
+        if surveys is None:
+            raise ValueError("surveys are required in order to use the MR feature.")
+        survey_name = surveys[0].name
+        image = batch["blend_images"][survey_name][idx]
+        wcs = batch["wcs"][survey_name]
+
+    # single-survey
+    else:
+        image = batch["blend_images"][idx]
+        wcs = batch["wcs"]
+
+    # run source extractor on the first band
+    band_image = image[0] if channel_indx == 0 else image[:, :, 0]
+    bkg = sep.Background(band_image)
+    catalog = sep.extract(band_image, sigma_noise, err=bkg.globalrms, segmentation_map=False)
+
+    # convert predictions to arcseconds
+    ra_coordinates, dec_coordinates = wcs.pixel_to_world_values(catalog["x"], catalog["y"])
+    ra_coordinates *= 3600
+    dec_coordinates *= 3600
+
+    # iterate over remaining bands and match predictions using KdTree
+    for band in range(1, image.shape[channel_indx]):
+        # run source extractor
+        band_image = image[band] if channel_indx == 0 else image[:, :, band]
+        bkg = sep.Background(band_image)
+        catalog = sep.extract(band_image, sigma_noise, err=bkg.globalrms, segmentation_map=False)
+
+        # convert predictions to arcseconds
+        ra_detections, dec_detections = wcs.pixel_to_world_values(catalog["x"], catalog["y"])
+        ra_detections *= 3600
+        dec_detections *= 3600
+
+        # convert to sky coordinates
+        c1 = SkyCoord(ra=ra_detections * units.arcsec, dec=dec_detections * units.arcsec)
+        c2 = SkyCoord(ra=ra_coordinates * units.arcsec, dec=dec_coordinates * units.arcsec)
+
+        # add new coordinates
+        if len(c1) > 0 and len(c2) > 0:
+            idx, d2d, d3d = c1.match_to_catalog_sky(c2)
+            d2d = d2d.arcsec
+
+            ra_coordinates = np.concatenate(
+                [ra_coordinates, ra_detections[d2d > matching_threshold]]
+            )
+            dec_coordinates = np.concatenate(
+                [dec_coordinates, dec_detections[d2d > matching_threshold]]
+            )
```

Comment:
just to make sure I understand, `ra_coordinates` are the detections from first band, and `ra_detections` on a separate band. You will never add the same center twice because you have the `matching_treshold` right?

Reply:
This looks correct to me, do you think it's worth checking with a unittest?

Reply:
Yes, exactly! `ra_coordinates` is the running "combined" list of coordinates. And  each time I get new detections, I make sure that they are no closer then `matching_threshold` before adding them to the "combined" list of coordinates.

Reply:
OK that makes sense to me, perhaps add a comment or two to this effect or in the docstring? I still find it a little hard to understand. Something like `# add new predictions for this band to catalog only if not close to existing centroid` ?

Reply:
now that I looked at the correct diff, this is fine now